src/lib/statistics/abstract_column_statistics.hpp

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <memory>
+
+#include "all_type_variant.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+class AbstractColumnStatistics;
+
+struct ColumnValueEstimate {
```

Comment:
I first thought this would estimate the value of a column, whatever that would be. Only when I saw `ColumnColumnEstimate`, I got an idea of what it might mean. Maybe `FilteredByValueEstimate`? The first `Column` seems obvious to me - what else would we filter?

I'm not too happy with `ColumnColumnEstimate` either - my first guess was that it was something needed for a join.

Reply:
those names are bad I'll come up with something better.